### DISCUSSION TITLE
AX.25 encoding on GNURadio 3.9

### DISCUSSION THREAD

**Discussion by SIARM8** [daniestevez/gr-satellites#320] [Q&A]
Dear all, Dear Daniel,

First of all thanks for the hard work with gr-satellites, it is really a great suite of sw for GNURadio.

I would like to set up a tx line using GNURADIO and my HackRF One, to be able to tx AX.25 packets, and then decode them using my Kenwood TH-D72A and using my RTL2838 UDB dongle managed by GQRX, with the stream decoded by DireWolf.
Transmitting with the Kenwood and receiving with the dongle work flawlessly.

I have GNURadio v3.9.2.0 (installed from 3.9 official APT repository), latest version of the gr-osmosdr available (compiled from source) and maint-3.9 version gr-satellites (compiled from source as well), latest version of GQRX (compiled from source). Pc is running Ubuntu 21.04.

By looking around and in literature, I have checked that the only external OOT module that encodes AX.25 packets (that I am aware of) seems to be gr-packetradio, which now seems to be an orphaned package, and it cannot be compiled if GNURadio v3.9 is installed (_make_ cmd throws out several errors, others had issues with 3.8 as well).

Is there any other OOT module that can be used to encode to an AX.25 packet? In gr-satellites, there is the HDLC framer, but the AX.25 module that should be placed before it is missing.

Thanks in advance for your precious suggestions.

Rick

**Comment by daniestevez** [comment]
Maybe the [AX.25 examples](https://github.com/daniestevez/gr-satellites/tree/master/examples/ax25) are useful.

What do you mean by "but the AX.25 module that should be placed before it is missing"? Do you need blocks to generate AX.25 headers? I'm not sure what is it exactly that you're missing.

**Reply by SIARM8** [reply]
Hello Daniel,

Thanks for the reply.
I had already studied the examples, but I couldn't progress with them.
Yes, I was referring to a block to process a string and add AX.25 headers (the link layer), then we will have the HDLC framer (the physical layer) that incapsulates the AX.25 frame. 
After it I have connected a PDU to tagged stream, scrambler, NZRI encode, Pack k Bits, GMSK Mod, RR and Osmocom Sink.

[This video](https://youtu.be/nrRXGQPRWpY?t=58) ( @ 58s) shows one of the blocks I am referring to in action:

As you have noticed, the block in the video includes the HDLC framing and G3RUH scrambling, which in gr-satellites are separate.

Based on your post [here](https://destevez.net/2016/06/kiss-hdlc-ax-25-and-friends/), I have installed ax25-tools, and now starting to configure the axports. Next step will be to try generating the AX.25 headers for a string and then transferring the AX.25 frame via UDP to GNURadio. It seems kind of complex but I will give it a try ;)

**Reply by daniestevez** [reply]
Just a suggestion. Perhaps it wouldn't be too difficult to make such a block to add AX.25 header using the [construct definitions](https://github.com/daniestevez/gr-satellites/blob/master/python/telemetry/ax25.py) in gr-satellites.
